src/api/middleware/auth.js

```diff
@@ -78,15 +78,26 @@ export const hasAdminPrivileges = async (req, res, next) => {
 };
 
 export const validToken = (req, res, next) => {
-    const decoded = verifyAndDecodeToken(req.params.token, config.jwt_secret);
+    try {
+        const decoded = verifyAndDecodeToken(req.params.token, config.jwt_secret);
 
-    if (!decoded) {
-        return next(new APIError(HTTPStatus.FORBIDDEN, ErrorTypes.FORBIDDEN, ValidationReasons.INVALID_TOKEN));
-    }
+        storeInLocals(req, {
+            token: decoded,
+        });
 
-    storeInLocals(req, {
-        token: decoded,
-    });
+        return next();
+    } catch (jwtErr) {
+        if (jwtErr.name === "TokenExpiredError") {
+            return next(new APIError(HTTPStatus.GONE, ErrorTypes.FORBIDDEN, ValidationReasons.EXPIRED_TOKEN));
```

Comment:
Why gone? This should just be forbidden, no? 

_"The HyperText Transfer Protocol (HTTP) 410 Gone client error response code indicates that access to the target resource is no longer available at the origin server and that this condition is likely to be permanent."_

Reply:
@FranciscoCardoso913 did you give an explanation regarding this?